chainer/initializers/__init__.py

```diff
@@ -26,8 +26,13 @@
 from chainer import types  # NOQA
 
 
-def generate_array(initializer, shape, xp, dtype=None, device=None):
-    # type: (types.AbstractInitializer, types.ShapeSpec, types.Xp, types.DTypeSpec, types.DeviceSpec) -> types.NdArray  # NOQA
+def generate_array(
+        initializer: types.AbstractInitializer,
+        shape: types.ShapeSpec,
+        xp: types.Xp,
+        dtype: types.DTypeSpec = None,
```

Comment:
Optional?

Reply:
I didn't change this annotation because `DTypeSpec` is `Union[Any]`.
https://github.com/chainer/chainer/blob/master/chainer/types.py#L30

a673492b5
But I agree to use `Optional[DTypeSpec]`.
I updated my PR, thanks a lot! ⛩